### ISSUE TITLE
[F6] bf16 output and weight support for the converter

### ISSUE THREAD

**Issue by yusiwen** [yusiwen/minfer#142]
F6 (#49) supports `--outtype f16` and `--outtype f32` in `minfer convert`.
`--outtype bf16` is refused:

```
minfer convert: --outtype bf16 is not supported; supported output types: f16,
f32 (a bf16 writer is a follow-up — f32 preserves every bf16 value exactly)
```

The engine also has no bf16 *weight* path (GGUF `GgmlType::BF16` is parsed but
neither `mat_mul_*` nor the device kernels dispatch it), so writing a bf16 GGUF
would produce a file the engine cannot run.

**What to build.**
- A bf16 output path in `src/convert.rs` (`bf16` in `OutType`, f32→bf16
  round-to-nearest-even writes) and a bf16 weight branch in the CPU matmul and
  embedding paths (decode bf16 = `f32::from_bits(bits << 16)`).
- Keep the HF-side bf16 *input* path as it is (it works).
- Optionally a device registration for bf16.

**Acceptance**
- `OutType::parse("bf16")` is accepted; the existing refusal test flips to an
  accepted-target test.
- A real-model gate loads a bf16-converted 0.5B, compares logits against the f16
  conversion within the stated bound (bf16 has 8 mantissa bits vs f16's 10, so
  the bound is looser and must be stated), and the greedy continuation matches.
- `docs/GGUF-TOOLING.md`'s supported/refused table is updated.
